### PR TITLE
add git plugin

### PR DESCRIPTION
**Pull request by Binyang2014**
```yaml
extras:
  com.microsoft.pai.runtimeplugin:
    - plugin: git
      parameters:
        repo_uri: <git repo>
        options:
        - <git clone options>
        clone_dir: <clone dir>
```

### CODE REVIEW

File: src/plugins/git/init.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    if "options" in parameters:
+        Repo.clone_from(parameters["repo_uri"],
+                        repo_local_path,
+                        multi_options=parameters["options"])
+    else:
+        Repo.clone_from(parameters["repo_uri"], repo_local_path)
+    if "clone_dir" in parameters:
+        plugin_helper.inject_commands([
+            "mkdir -p {}".format(parameters["clone_dir"]),
+            "mv -f {}/* {}".format(repo_local_path, parameters["clone_dir"])
```
Comment (hzy46) [inline src/plugins/git/init.py:58]: Why use `mv` here? If `clone_dir` is in `parameters`, we can use `Repo.clone_from(parameters["repo_uri"], parameters["clone_dir"])`

File: src/plugins/git/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+## Schema
+```yaml
+extras:
+  com.microsoft.pai.runtimeplugin:
+    - plugin: git
+      parameters:
+        repo_uri: <git repo>
+        options:
+        - <git clone options>
+        clone_dir: <clone dir>
```
Comment (hzy46) [inline src/plugins/git/README.md:15]: If the user doesn't give `clone_dir`, the plugin will use `/code` (or `~/code`? I'm not quite sure) as the default dir. Please document the behavior.

File: src/plugins/git/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+```
```
Comment (hzy46) [inline src/plugins/git/README.md:16]: Also, what will happen if the default dir or `clone_dir` exists before git cloning?

File: src/plugins/git/init.py (hunk 1 above)
Comment (Binyang2014) [reply src/plugins/git/init.py:58]: No, clone happens in init-container. then we need to mv repo from init-container to app-container.

File: src/plugins/git/README.md (hunk 2 above)
Comment (Binyang2014) [reply src/plugins/git/README.md:15]: Will use path `/usr/local/pai/code`, will doc this

File: src/plugins/git/README.md (hunk 3 above)
Comment (Binyang2014) [reply src/plugins/git/README.md:16]: If it has file in same name. will overwrite it. Will doc this. Customer should be care of this case

Comment (hzy46) [reply src/plugins/git/README.md:16]: Can we throw an error if this happens? I think overwriting user's data may not be a good behavior.

File: src/plugins/git/README.md (hunk 2 above)
Comment (Binyang2014) [reply src/plugins/git/README.md:15]: Added doc, thx.

File: src/plugins/git/README.md (hunk 3 above)
Comment (Binyang2014) [reply src/plugins/git/README.md]: Added doc, thx.

Comment (Binyang2014) [reply src/plugins/git/README.md]: > Can we throw an error if this happens? I think overwriting user's data may not be a good behavior.

I think this parameter is set by user, user should aware this. It's user responsibility to take care of this parameter

**Review by hzy46 — APPROVED** [review]

File: src/plugins/git/README.md (hunk 3 above)
Comment (Binyang2014) [reply src/plugins/git/README.md]: Change to fail the command and throw error. Which respect git default behavior

**Pull request merged**